internal/pipeline/executor.go

```diff
@@ -0,0 +1,271 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipeline
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"go.uber.org/zap"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// Executor hydrates a given pkg.
+type Executor struct {
+	PkgPath string
+	Logger  *zap.Logger
```

Comment:
I'm thinking the logger should be passed through as a function parameter instead of being set on the structs. Maybe as a value on a context?

Reply:
The downside of passing in context will be that every function will have to take a context, even the helper functions.
I will check klog and see if it can be a good alternative without requiring passing logger.